proposals/NNNN-vehicle-type-filter.md

```diff
@@ -0,0 +1,435 @@
+        trim="GT"/>
+</resource>
+```
+3. If an app defines a `vehicle-type` element, then it should always have a `make` attribute, all other attributes are optional. However, if the app developers want to use `model year` or `trim`, they should define `make` and `model` attributes as well. The proxy will check only the defined attributes. The below example shows a valid vehicle type resource file.
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<resource>
+<!-- Vehicle filter for vehicle make-->
+    <vehicle-type
+        make="Ford"/>
+</resource>
+```
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<resource>
+<!-- Vehicle filter for vehicle make and model-->
+    <vehicle-type
+        make="Ford"
+        model="Mustang"/>
+</resource>
+```
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<resource>
+<!-- Vehicle filter for vehicle make, model and model year-->
+    <vehicle-type
+        make="Ford"
+        model="Mustang"
+        modelYear="2019"/>
+</resource>
+```
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<resource>
+<!-- Vehicle filter for vehicle make, model and trim-->
+    <vehicle-type
+        make="Ford"
+        model="Mustang"
+        trim="GT"/>
+</resource>
+```
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<resource>
+<!-- Vehicle filter for vehicle make-->
+    <vehicle-type
+        make="Ford"/>
+    <vehicle-type
+        make="OEM1"/>
+    <vehicle-type
+        make="OEM2"/>
+</resource>
+```
+4. If the supported vehicle type list is not defined, the proxy can start `SDL router service` for all SDL enabled IVI system.
+
+#### Meta-data for SDL router service
+
+1. This resource file needs to be referenced in the manifest file of the project as `meta-data` for `Sdl router service`.
+2. When the proxy receives `Get vehicle type ACK`, it will check if the vehicle type information is defined in the provided supported vehicle type list.
+3. If the vehicle filter is not defined, the proxy can skip the check. In this case, the SDL router service is supported for all SDL enabled system.
+
+#### Determine Vehicle type info
+
+1. In the current implementation, all SDL apps need to start version negotiations and register themselves on SDL enabled system. This proposal recommends starting RPC service by the SDL router service to get vehicle type information from SDL enabled IVI system before notifying clients about SDL connection.
+2. If the protocol version supports `Get vehicle type`, SDL router service will send the `Get vehicle type` protocol message.
+3. If the SDL router service supports the connected IVI system, then the SDL router service will notify the client and provide vehicle type information. The router service will also transfer RPC session information to application hosting router service, since, every app is responsible for version negotiations and starting RPC session, the host application will use the same session information to communicate further with the SDL system. 
```

Comment:
```suggestion
3. If the SDL router service supports the connected IVI system, then the SDL router service will notify the client and provide vehicle type information. The router service will also transfer RPC session information to the application hosting the router service. Since every app is responsible for version negotiations and starting RPC session, the host application will use the same session information to communicate further with the SDL system. 
```